src/utils/misc.py

```diff
@@ -0,0 +1,13 @@
+def dict_reverse_lookup(dictionary : dict, value : str) -> str:
```

Comment:
Drop the `str` type-annotations here, since the dictionary values could in theory be any type.

Reply:
For the record, you can use
```py
from typing import TypeVar
T = TypeVar('T')

def dict_reverse_lookup(dictionary : dict, value : T) -> T:
    ...
```
but in this case I'm not running any type checkers, I'm just using the types to help document what the functions are doing, and ensure return type consistency -- so in this case, I'd ask you to drop them.